src/trait_manager/model/Trait.ts

```diff
@@ -85,6 +102,32 @@ export default class Trait extends Model<TraitProperties> {
     }
   }
 
+  listenToDataVariable(dataVar: TraitDataVariable) {
+    const { em } = this;
+    const { path } = dataVar.attributes;
+    const normPath = stringToPath(path || '').join('.');
+    const dataListeners: DataVariableListener[] = [];
+    const prevListeners = this.dataListeners || [];
+
+    prevListeners.forEach((ls) => this.stopListening(ls.obj, ls.event, this.updateValueFromDataVariable));
+
+    dataListeners.push({ obj: dataVar, event: 'change:value' });
+    dataListeners.push({ obj: em, event: `${DataSourcesEvents.path}:${normPath}` });
+
+    dataListeners.forEach((ls) =>
+      this.listenTo(ls.obj, ls.event, () => {
+        const dr = dataVar.getDataValue();
+        this.updateValueFromDataVariable(dr);
+      }),
+    );
+    this.dataListeners = dataListeners;
+  }
```

Comment:
I see that we're using the same pattern in `StyleableModel.ts` and `ComponentDataVariableView.ts` (this one is actually more complete as here you don't update the value when for example a record is removed).

Maybe we should consolidate this logic in one single place? eg. `data_sources/model/DataVariable.ts` is the base class, `data_sources/model/TraitDataVariable.ts` might extend some logic if necessary

Reply:
Listener logic is now inside the new class `DataVariableListenerManager`.